boa_icu_provider/src/lib.rs

```diff
@@ -1,3 +1,21 @@
+//! Boa's **boa_icu_provider** crate contains the default data provider used by its `Intl`
+//! implementation.
+//!
+//! # Crate Overview
+//! This crate exports the function [`blob`], which contains an extensive dataset of locale data to
```

Comment:
suggestion: y'all might benefit from using the "baked" backend for datagen which generates a very efficient Rust crate.

Reply:
Thought about using that, but when I try to `include!(concat!())` the generated module, Rust-analyzer goes crazy and starts throwing errors on the whole workspace.

Reply:
Yeah it's a very large crate full of const data that rust-analyzer doesn't like.

The trick we use is `include!(concat!(env!("CARGO_MANIFEST_DIR"), "/data/baked/mod.rs"));`, and we have the generated code live in `data` instead of `src`

Reply:
Yeah, I used the exact same technique, but R-A insists on trying to analyze the generated file :(

Reply:
The weird thing is that when I open one of the generated files, R-A marks the `// @generated` with something like "file not included in analysis"

Reply:
Does this still happen if the files get renamed to `.rs_include` or something?

Reply:
Ah, didn't think of doing that. I'll try it when I return home.

Reply:
we can make it a datagen option to do that so it's easier, if that works

there really needs to be a way to tell r-a to bugger off

Reply:
Yeah. Though, it'll also need to be accompanied with a way to declare the public exports of a file, or else R-A will complain about missing symbols.

Reply:
Tried to rename the included files with no success 🙁
We'll use binary blobs in the meantime, and I'll try to keep up with R-A to switch to baked data when the issue gets fixed. Are there any related issues I should subscribe to?

Reply:
We haven't filed any because so far it's worked for us. I think it might be because we use Cargo features.

Y'all should go ahead and file something and cc me.

Reply:
Done!